apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx

```diff
@@ -103,14 +99,43 @@ export function PoolParamsForm() {
     resolver: zodResolver(schemaMapper[poolType]),
     mode: "onSubmit",
   });
-  const { register, setValue, getValues, clearErrors } = form;
+  const {
+    register,
+    setValue,
+    getValues,
+    clearErrors,
+    formState: { errors },
+  } = form;
 
   const onSubmit = (data: FieldValues) => {
     setIsGraphLoading(true);
-    setCustomData(data as AnalysisData);
-    setInitialData(data as AnalysisData);
-    setAnalysisTokenByIndex(0);
-    setCurrentTabTokenByIndex(1);
+
+    const additionalParams =
+      poolType === PoolTypeEnum.GyroE
+        ? {
+            dSq: initialData.poolParams?.dSq,
+            tauAlphaX: initialData.poolParams?.tauAlphaX,
+            tauAlphaY: initialData.poolParams?.tauAlphaY,
+            tauBetaX: initialData.poolParams?.tauBetaX,
+            tauBetaY: initialData.poolParams?.tauBetaY,
+            u: initialData.poolParams?.u,
+            v: initialData.poolParams?.v,
+            w: initialData.poolParams?.w,
+            z: initialData.poolParams?.z,
+          }
+        : {};
```

Comment:
is there more data than you need within initialData.poolParams? Otherwise you could try to use ducktyping:
```suggestion
      poolType === PoolTypeEnum.GyroE && initialData.poolParams?.dSq
        ? initialData.poolParams
        : {};
```

Reply:
PYC refactored this to be:

```
    const dataWithPoolType = {
      poolParams: Object.fromEntries(
        inputMapper[poolType].map((input) => [input.name, data[input.name]])
      ),
      tokens: initialData.tokens,
      poolType,
    };
```